code/modules/mining/lavaland/necropolis_chests.dm

```diff
@@ -9,146 +9,82 @@
 	icon_state = "necro_crate"
 	resistance_flags = LAVA_PROOF | FIRE_PROOF | ACID_PROOF
 	door_anim_time = 0
-
-/obj/structure/closet/crate/necropolis/tendril
-	desc = "It's watching you suspiciously."
 	///prevents bust_open to fire
 	integrity_failure = 0
 	/// var to check if it got opened by a key
 	var/spawned_loot = FALSE
 
-/obj/structure/closet/crate/necropolis/tendril/Initialize(mapload)
+/obj/structure/closet/crate/necropolis/Initialize(mapload)
 	. = ..()
 	RegisterSignal(src, COMSIG_PARENT_ATTACKBY, .proc/try_spawn_loot)
 
-/obj/structure/closet/crate/necropolis/tendril/proc/try_spawn_loot(datum/source, obj/item/item, mob/user, params) ///proc that handles key checking and generating loot
+/obj/structure/closet/crate/necropolis/proc/try_spawn_loot(datum/source, obj/item/item, mob/user, params)
```

Comment:
There should just be a list of loot typepaths and use pickfromlistweighted or whatever the proc is, rather than this massive switch on every single subtype

Reply:
What massive switch are you talking about? The massive switch only applies to necropolis/tendril, not all necropolis/